Implement default_value_from in collector

rafamanzo opened the pull request:
It verifies parameters then delegates to the parsers.

Signed-off-by: Rafael Reggiani Manzo rr.manzo@protonmail.com

danielkza committed:
Implement default_value_from in collector

It verifies parameters then delegates to the parsers.

Signed-off-by: Rafael Reggiani Manzo <rr.manzo@protonmail.com>

diegoamc commented on spec/collector_spec.rb:
```diff
@@ -87,5 +87,33 @@
+                                                                                     'Metric configuration does not belong to MetricFu')
+        end
+      end
+    end
```
What about the case where `metric_configuration` is `nil`?

rafamanzo committed:
Remove unecessary check for nil metric

The default value retrieval was assuming that metrics within a
MetricConfiguration could be nil which is not permited by
`MetricConfiguration#metric=`.

Signed-off-by: Daniel Miranda <danielkza2@gmail.com>

rafamanzo commented:
This was part of mezuro/kalibro_processor#193
